java/ql/lib/semmle/code/java/dispatch/DispatchFlow.qll

```diff
@@ -167,6 +167,20 @@ private module TypeTrackingSteps {
       def.(BaseSsaUpdate).getDefiningExpr().(VariableAssign).getSource() = n1.asExpr()
       or
       def.(BaseSsaImplicitInit).isParameterDefinition(n1.asParameter())
+      or
+      exists(PatternCase pc |
+        pc.getPattern().asBindingPattern() = def.(BaseSsaUpdate).getDefiningExpr() and
+        (
+          pc.getSwitch().getExpr() = n1.asExpr()
+          or
+          pc.getSwitchExpr().getExpr() = n1.asExpr()
```

Comment:
Same here: These couple of added cases should be superfluous once they are included in `VariableAssign.getSource()`.